src/main/java/com/google/sps/servlets/UserServlet.java

```diff
@@ -72,30 +72,37 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     for (Cookie c:cookies) {
       if (c.getName().equals("user"))
         username = c.getValue();
-      if (c.getName().equals("id"))
-        id = Long.parseLong(c.getValue());
+      if (c.getName().equals("id")) {
+            System.out.println("ID ----> "+ c.getValue());
+            id = Long.parseLong(c.getValue());
+      }
     }
 
+    Key key = KeyFactory.createKey("user", id);
+
     Query query = new Query("user").setFilter(
-        CompositeFilterOperator.and(FilterOperator.EQUAL.of("username", username), FilterOperator.EQUAL.of(Entity.KEY_RESERVED_PROPERTY, id)));
+        CompositeFilterOperator.and(FilterOperator.EQUAL.of("username", username), FilterOperator.EQUAL.of(Entity.KEY_RESERVED_PROPERTY, key)));
     
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery result = datastore.prepare(query);
 
     Entity person = result.asSingleEntity();
-
-    String name = (String) person.getProperty("name");
-    String email = (String) person.getProperty("email");
-    String organization = (String) person.getProperty("organization");
-    String number = (String) person.getProperty("contactNum");
-
-    User user = new User (username, name, email, number, organization);
-
-    Gson gson = new Gson();
-
-    response.setContentType("application/json; charset=UTF-8");
-    response.setCharacterEncoding("UTF-8");
-    response.getWriter().println(gson.toJson(user));
+    if (person == null) {
+        User user = new User(null, null, null, null, null);
+        Gson gson = new Gson();
+        response.setContentType("application/json; charset=UTF-8");
+        response.setCharacterEncoding("UTF-8");
+        response.getWriter().println(gson.toJson(user));
+    } else {
+        String name = (String) person.getProperty("name");
+        String email = (String) person.getProperty("email");
+        String organization = (String) person.getProperty("organization");
+        String number = (String) person.getProperty("contactNum");
+        User user = new User (username, name, email, number, organization);
+        Gson gson = new Gson();
+        response.setContentType("application/json; charset=UTF-8");
+        response.setCharacterEncoding("UTF-8");
+        response.getWriter().println(gson.toJson(user));
```

Comment:
Starting from the line Gson to the line with response.getWriter(), since all those lines are the same in either condition, you could probably just call them once, at the bottom of the if/else condition.